specification/hybridnetwork/resource-manager/Microsoft.HybridNetwork/preview/2022-01-01-preview/networkFunction.json

```diff
@@ -0,0 +1,570 @@
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The URL to get the next set of results."
+        }
+      },
+      "description": "Response for network function API service call."
+    },
+    "NetworkFunctionUserConfiguration": {
+      "type": "object",
+      "properties": {
+        "roleName": {
+          "type": "string",
+          "description": "The name of the network function role."
+        },
+        "userDataParameters": {
+          "$ref": "./common.json#/definitions/UserDataParameters",
+          "description": "The user data parameters from the customer."
+        },
+        "networkInterfaces": {
+          "type": "array",
+          "items": {
+            "$ref": "./common.json#/definitions/NetworkInterface"
+          },
+          "description": "The network interface configuration."
+        },
+        "osProfile": {
+          "type": "object",
+          "properties": {
+            "customData": {
+              "type": "string",
+              "description": "Specifies a base-64 encoded string of custom data. The base-64 encoded string is decoded to a binary array that is saved as a file on the virtual machine. The maximum length of the binary array is 65535 bytes. <br><br> **Note: Do not pass any secrets or passwords in customData property** <br><br> This property cannot be updated after the VM is created. <br><br> customData is passed to the VM to be saved as a file. For more information see [Custom Data on Azure VMs](https://azure.microsoft.com/en-us/blog/custom-data-and-cloud-init-on-windows-azure/) <br><br> For using cloud-init for your Linux VM, see [Using cloud-init to customize a Linux VM during creation](https://docs.microsoft.com/azure/virtual-machines/virtual-machines-linux-using-cloud-init?toc=%2fazure%2fvirtual-machines%2flinux%2ftoc.json)"
+            }
+          },
+          "description": "Specifies the operating system settings for the role instance."
+        }
+      },
+      "description": "The network function user configuration."
+    },
+    "ExecuteRequestParameters": {
+      "type": "object",
+      "properties": {
+        "serviceEndpoint": {
+          "type": "string",
+          "description": "The endpoint of service to call."
+        },
+        "requestMetadata": {
+          "$ref": "#/definitions/RequestMetadata",
+          "description": "The request metadata."
+        }
+      },
+      "required": [
+        "serviceEndpoint",
+        "requestMetadata"
+      ],
+      "description": "Payload for execute request post call."
+    },
+    "RequestMetadata": {
+      "type": "object",
+      "properties": {
+        "relativePath": {
+          "type": "string",
+          "description": "The relative path of the request."
+        },
+        "httpMethod": {
+          "type": "string",
+          "description": "The http method of the request."
```

Comment:
consider making this an extensible enum with the httpmethod params

Reply:
Thank you for the suggestion. Will consider this idea.

Reply:
Changed it to enum.